### PR TITLE
MatthewKuKanich/findmyflipper_1.0

### PR DESCRIPTION
**Pull request by MatthewKuKanich**
# Application Submission

- The FindMy Flipper app turns the FlipperZero into an effective tracking device, compatible with both Apple AirTags and Samsung SmartTags. It uses the BLE beacon to broadcast, allowing users to clone existing tags, generate OpenHaystack key pairs for Apple's FindMy network, and customize beacon intervals and transmit power. 

# Extra Requirements 

- dev branch (for extra ble beacon)

# Author Checklist (Fill this out)

- [X] I've read the [contribution guidelines](../blob/HEAD/documentation/Contributing.md) and my PR follows them
- [X] I own the code I'm submitting or have code owner's permission to submit it
- [x] I [have validated](../blob/HEAD/documentation/Contributing.md#validating-manifest) the manifest file(s) with `python3 tools/bundle.py --nolint applications/CATEGORY/APPID/manifest.yml bundle.zip`

# Reviewer Checklist (Don't fill this out)

- [x] Bundle is valid
- [x] There are no obvious issues with the source code
- [x] I've ran this application and verified its functionality

### CODE REVIEW

**Comment by MatthewKuKanich** [comment]
validated with ```python tools/bundle.py --nolint applications/Bluetooth/findmy/manifest.yml bundle.zip``` after pointing towards repos .git and adding subdirectory to manifest to lock onto the OFW branch

File: applications/Bluetooth/findmy/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+sourcecode:
+  type: git
+  location:
+    origin: https://github.com/MatthewKuKanich/FindMyFlipper.git
+    commit_sha: 7e6b8f386b8b6a09e4d4e8361f85a230e7a8bfc4
+    subdir: tree/ofw/findmy
```
Comment (hedger) [inline applications/Bluetooth/findmy/manifest.yml:6]: Wrong path.

**Review by hedger — DISMISSED** [review]
Please actually run validator script when submitting changes.

**Comment by MatthewKuKanich** [comment]
> Please actually run validator script when submitting changes.

I did, I apologize for the confusion. I believed that the issue was checking out the branch (because that failed before) and it passed that but failed building temp. I thought that was on my end and since it reached the build stage that it was good to go. But I did actually run the validator script. On a side note, will it build to the dev branch? It needs dev to use this feature. It is integral in having a working OFW version of the app as that is what works as the airtag beacon. Everything else passes. 
Logs: 
```
PS C:\Development\FlipperZero\flipper-application-catalog> python tools/bundle.py --nolint applications/Bluetooth/findmy/manifest.yml bundle.zip
23:58:09.777 [I] Deploying SDK for f7
23:58:09.778 [I] Fetching version info for UpdateChannel.RELEASE from https://update.flipperzero.one/firmware/directory.json
23:58:09.944 [I] Using version: 0.98.3
23:58:09.944 [I] uFBT SDK dir: C:\Users\Matthew\.ufbt\current
23:58:09.945 [I] SDK is up-to-date
23:58:10.188 [I] Working in 'C:\Users\Matthew\AppData\Local\Temp\tmpfbkico0w'
23:58:10.191 [I] Loaded app manifest from applications\Bluetooth\findmy\manifest.yml
23:58:10.191 [I] Cloning https://github.com/MatthewKuKanich/FindMyFlipper.git to C:\Users\Matthew\AppData\Local\Temp\tmp52o7j6dl
Cloning into 'C:\Users\Matthew\AppData\Local\Temp\tmp52o7j6dl'...
remote: Enumerating objects: 207, done.
remote: Counting objects: 100% (75/75), done.
remote: Compressing objects: 100% (51/51), done.
Receiving objects:  92% (191/207)sed 30 (delta 23), pack-reused 132
Receiving objects: 100% (207/207), 134.66 KiB | 12.24 MiB/s, done.
Resolving deltas: 100% (96/96), done.
23:58:11.368 [I] Cloned. Checking out commit
Note: switching to '4d1ca55f2a3b35efe339e10955985a21f5f3508f'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 4d1ca55 change naming scheme and print save directory
23:58:11.425 [I] Checked out commit 4d1ca55f2a3b35efe339e10955985a21f5f3508f. Updating submodules
23:58:12.081 [I] Moving C:\Users\Matthew\AppData\Local\Temp\tmp52o7j6dl\FindMyFlipper to C:\Users\Matthew\AppData\Local\Temp\tmpfbkico0w\code
23:58:12.082 [I] Building
In file included from C:\Users\Matthew\AppData\Local\Temp\tmpfbkico0w\code\findmy_state.c:1:
C:\Users\Matthew\AppData\Local\Temp\tmpfbkico0w\code\findmy_state.h:3:10: fatal error: extra_beacon.h: No such file or directory
    3 | #include <extra_beacon.h>
      |          ^~~~~~~~~~~~~~~~

**Comment by hedger** [comment]
@MatthewKuKanich Beacon API is available in both dev & RC channels, abd is coming to release channel soon. Please run `ufbt update -c rc` before bundler script.

**Comment by MatthewKuKanich** [comment]
> @MatthewKuKanich Beacon API is available in both dev & RC channels, abd is coming to release channel soon. Please run `ufbt update -c rc` before bundler script.
> 
> Anyway, folder `tree/ofw/findmy` does not exist in the repo, and the log you posted references commit 4d1ca55f2a3b35efe339e10955985a21f5f3508f - which is different from the one in your PR.

Yes, I didn't want to send in another build until I knew I had it working so the commit reference is old. Should be up-to-date in the latest version. And succeeded the validation script.

**Review by doomwastaken — APPROVED** [review]

**Comment by MatthewKuKanich** [comment]
3 issues were opened on my repo, I'm not sure what the issue actually is. https://github.com/MatthewKuKanich/FindMyFlipper/issues/20

**Comment by MatthewKuKanich** [comment]
> Sorry for the troubles, we are trying to fix issue creation flow, that we only notify on major api upgrade (under discussion now, but basically "as rare as possible") and if you close and issue, new one should not be created again.
> 
> Once again, sorry for the troubles, you can close all those issues, we are not tracking them, just notifying authors on failed builds

I don't know why the build failed. It builds correctly on my local machine, with the validator script, and in your PR Github action. It passed everything and was merged. Did something happen?

**Pull request merged**